packages/fx-core/src/core/middleware/videoFilterAppBlocker.ts

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+"use strict";
+
+import { NextFunction } from "@feathersjs/hooks";
+import { Inputs, err, Func } from "@microsoft/teamsfx-api";
+import { isVideoFilterProject } from "../../common/tools";
+import { VideoFilterAppRemoteNotSupportedError } from "../error";
+import { CoreHookContext } from "../types";
+
+const userTasksToBlock: Func[] = [
+  // Teams: Add features
+  {
+    namespace: "fx-solution-azure",
+    method: "addFeature",
+  },
+  // Teams: Zip Teams metadata package
+  {
+    namespace: "fx-solution-azure",
+    method: "buildPackage",
+  },
+  // Teams: Validate manifest file
+  {
+    namespace: "fx-solution-azure",
+    method: "validateManifest",
+  },
+];
+
+async function shouldBlockExecution(ctx: CoreHookContext): Promise<boolean> {
+  const inputs = ctx.arguments[ctx.arguments.length - 1] as Inputs;
+  if (!inputs.projectPath) {
+    return false;
+  }
+
+  if (ctx.method === "executeUserTask") {
```

Comment:
Looks it only blocks the defined executed user tasks, where's the logic to block operation like provision, deploy?

Reply:
Normal lifecycle methods that have this middleware are blocked by default. But some operations in `executeUserTask` should be blocked and some don't. So we need this list only for `executeUserTask`.

![image](https://user-images.githubusercontent.com/9698542/203253330-3811cede-6564-416f-b557-f251f1f1375d.png)